src/main/java/Duke.java

```diff
@@ -1,10 +1,261 @@
+        }
+
+        String greet = line +
+                "Hello! I'm Duke.\n" +
+                "To add general tasks to the todo list, simply type in the task and press enter.\n" +
+                "You can specify the type of tasks by using " +
+                "'todo' / 'deadline' / 'event' keyword before the task description.\n" +
+                "You can keep track of deadlines by typing 'deadline' " +
+                "followed by task description + '/by' to specify the time of the deadline.\n" +
+                "You can also add events to the list by typing 'event' " +
+                "followed by task description + '/from' + starting time of event + '/to' ending time of event.\n" +
+                "Type 'list' to view the current todo list.\n" +
+                "Type 'mark' and task event index to mark the task as done " +
+                "and type 'unmark' and index to undo the task.\n" +
+                "Type 'bye' to exit Duke.\n" +
+                line;
+        System.out.println(greet);
+
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
+            String userInput = "";
+            if (in.hasNextLine()) {
+                userInput = in.nextLine().trim();
+            } else {
+                break;
+            }
+            try {
+                if (userInput.equals("bye")) {
+                    System.out.println("Thank you for using Duke. Hope to see you soon!");
+                    break;
+                } else if (userInput.equals("list")) {
+                    printTaskList();
+                } else if (userInput.startsWith("mark")) {
+                    markTask(userInput);
+                } else if (userInput.startsWith("unmark")) {
+                    unmarkTask(userInput);
+                } else if (userInput.startsWith("todo")) {
+                    try {
+                        createTodo(userInput);
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.println("Description of a todo cannot be empty. Adding todo failed.\n" + line);
+                    }
+                } else if (userInput.startsWith("deadline")) {
+                    try {
+                        createDeadline(userInput);
+                    } catch (StringIndexOutOfBoundsException e) {
+                        System.out.println("Insufficient input for deadline. Adding deadline failed.\n" + line);
+                    }
+                } else if (userInput.startsWith("event")) {
+                    try {
+                        createEvent(userInput);
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.println("Insufficient input for event. Adding event failed.\n" + line);
+                    }
+                } else if (userInput.startsWith("delete")) {
+                    deleteTask(userInput);
+                } else {
+                    throw new TaskTypeException();
+                }
```

Comment:
Likewise, you might want to consider abstracting this section of your progarm as it is quite long and contains deeply-nested code.